project/cmake/modules/LDGOLD.cmake

```diff
@@ -1,4 +1,8 @@
-option(ENABLE_LDGOLD "Use GNU gold linker" ON)
+if(CMAKE_BUILD_TYPE STREQUAL Release)
+  option(ENABLE_LDGOLD "Use GNU gold linker" ON)
+else()
+  option(ENABLE_LDGOLD "Use GNU gold linker" OFF)
+endif()
```

Comment:
Well, then you can just disable it. It mostly does make sense on debug builds. Because it can split the debug info into extern files for better link performance. Can you please add a note to the commit message why this is needed, otherwise such changes have a very high chance of being accidentally removed again.